war/src/main/java/executable/LogFileOutputStream.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2008, Sun Microsystems, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package executable;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.FilterOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.UncheckedIOException;
+import sun.misc.Signal;
```

Comment:
How does Java 17 not complain about this?

Reply:
When compiling with Java 9+, this doesn't work with `release=8`, but it does work with `source=1.8` and `target=1.8`, which is why we clear the `release` flag in `pom.xml`. See [JENKINS-68021](https://issues.jenkins.io/browse/JENKINS-68021).

Reply:
```suggestion
import sun.misc.Signal; // JENKINS-68021
```
Maybe we can just drop this functionality. Out of the small number of users who even use the `--logfile` option to begin with, how many know or care that you can

```bash
killall -ALRM jenkins
```

to rotate it? https://github.com/jenkinsci/jenkins/blob/fee5a39b73ea2bef14393d4be4fae83f1a0978a5/war/src/main/java/executable/Main.java#L297 does not mention this. Is there some native package which really needs it? (Normally service wrappers in native packages manage log redirection using standard service tools which would have this feature among many others.)

Reply:
Because I am seeing it now for the first time, and according to Jira discussion is tangentially related to this PR due to POM tricks. Just suggesting at least a reference to the issue you filed whose existence would not otherwise be obvious.

Reply:
I added commit 36de63e28f to document the workaround. If you want to discuss changing this long-standing behavior in existing code, fine, but please open a new ticket or mailing list thread and keep it out of this PR.